renderer/src/hooks/useCurrentTransactionStatus.tsx

```diff
@@ -0,0 +1,34 @@
+import { useEffect, useRef, useState } from 'react'
+import { FILTransactionProcessing } from '../../../shared/typings'
+
+const useCurrentTransactionStatus = (transaction?: FILTransactionProcessing) => {
+  const [status, setStatus] = useState<'none' | 'processing' | 'complete'>('none')
```

Comment:
I should have brought this up earlier, but this type conflicts with that one, which we're already using throughout the codebase:

```ts
export type FILTransactionStatus = 'succeeded' | 'processing' | 'failed'
```

Could we please unify these types? `complete -> succeeded` is an obvious match, what about `none` though?

Reply:
I need a status for when there is no transaction. Maybe `idle` is better?

Reply:
Why would a transaction have no transaction? I don't understand this state. As soon as you create a transaction it's processing.

Reply:
what about returning `FILTransactionStatus || null`?

Reply:
The point of the hook is to know if there is a transaction processing, and if there is, what state it's in. I was combining that into a single state, but if you prefer I could separate the two, and return 
```
{
 status: FILTransactionStatus
 transaction?: FILTransaction
}
```

Reply:
Hadn't seen the latest comment, that sounds good to me

Reply:
Simplified it a bit to use the tx status directly, instead of keeping separate state for it